File: backend-nestjs/src/dto/requests/gallery-image.dto.ts
import { IsNotEmpty, IsOptional, IsString } from 'class-validator';

export class GalleryImageDto {
  @IsNotEmpty()
  @IsString()
  url!: string;

  @IsOptional()
  @IsString()
  alt?: string;

  @IsOptional()
  @IsString()
  caption?: string;
}

File: backend/src/main/java/com/contentplatform/backend/application/dto/GalleryImageDto.java
package com.contentplatform.backend.application.dto;

public class GalleryImageDto {
    private final String url;
    private final String alt;
    private final String caption;

    public GalleryImageDto(String url, String alt, String caption) {
        this.url = url;
        this.alt = alt;
        this.caption = caption;
    }

    public String getUrl() {
        return url;
    }

    public String getAlt() {
        return alt;
    }

    public String getCaption() {
        return caption;
    }
}
